### PR TITLE
[Runtime] Fixed several memory leaks

### PR DESCRIPTION
**Pull request by Dunati**

### CODE REVIEW

**Comment by akoeplinger** [comment]
build

File: mono/metadata/appdomain.c [hunk 1]
```diff
@@ -1448,6 +1449,7 @@ get_shadow_assembly_location_base (MonoDomain *domain, MonoError *error)
 	char *cache_path, *appname;
 	char *userdir;
 	char *location;
+	char *username;
```
Comment (evincarofautumn) [inline mono/metadata/appdomain.c]: Dead code?

File: mono/metadata/sgen-mono.c [hunk 2]
```diff
@@ -3005,7 +3005,7 @@ mono_gc_base_init (void)
 void
 mono_gc_base_cleanup (void)
 {
-	sgen_thread_pool_shutdown ();
+	sgen_thread_pool_shutdown ();
```
Comment (schani) [inline mono/metadata/sgen-mono.c]: Whitespace damage.

File: mono/metadata/icall.c [hunk 3]
```diff
@@ -9111,7 +9119,7 @@ mono_register_jit_icall_full (gconstpointer func, const char *name, MonoMethodSi
 
 	info = g_new0 (MonoJitICallInfo, 1);
 	
-	info->name = name;
+	info->name = g_strdup(name);
```
Comment (evincarofautumn) [inline mono/metadata/icall.c]: I take it we were assuming the wrong ownership here.

File: mono/metadata/loader.c [hunk 4]
```diff
@@ -2461,6 +2468,8 @@ mono_method_signature_checked (MonoMethod *m, MonoError *error)
 	mono_memory_barrier ();
 	if (!m->signature)
```
Comment (evincarofautumn) [inline mono/metadata/loader.c]: We can get rid of some double-negatives here.

```
if (m->signature) {
    if (!can_cache_signature && !img)  // or: !(can_cache_signature && img)
        g_free (signature);
} else {
    m->signature = signature;
}
```

File: mono/metadata/sgen-mono.c (hunk 2 above)
Comment (evincarofautumn) [inline mono/metadata/sgen-mono.c]: Whitespace.

File: mono/metadata/appdomain.c (hunk 1 above)
Comment (Dunati) [inline mono/metadata/appdomain.c]: yes, the call to g_get_user_name leaks on windows and this was part of a different fix, will remove.

**Comment by Dunati** [comment]
Okay, updated on based on feedback and removed a couple changes that caused trouble with some of the tests.

File: mono/metadata/icall.c (hunk 3 above)
Comment (Dunati) [inline mono/metadata/icall.c]: Some codepaths to mono_register_jit_icall_full frees the string elsewhere and some do not. The duplicated string is freed in mono_icall_cleanup.

**Comment by Dunati** [comment]
ug, i think i messed up the history when trying to update the branch, closing this one in favor of  https://github.com/mono/mono/pull/3309

**Pull request closed without merging**